src/tim/vx/ops/conv2d_test.cc

```diff
@@ -940,3 +942,415 @@ TEST(Conv2d, shape_4_2_1_2_float32_InputAndFilterSameWidthHeightTest) {
   EXPECT_TRUE(output_tensor->CopyDataFromTensor(output.data()));
   EXPECT_EQ(golden, output);
 }
+
+TEST(Conv2d, shape_4_2_1_2_uint8_QuantizedTest1) {
+  auto ctx = tim::vx::Context::Create();
+  auto graph = ctx->CreateGraph();
+  tim::vx::ShapeType input_shape({4, 2, 1, 2});  //whcn
+  tim::vx::ShapeType weight_shape({2, 2, 1, 3}); //whio
+  tim::vx::ShapeType bias_shape({weight_shape[3]});
+  tim::vx::ShapeType output_shape(
+      {2, 1, weight_shape[3], input_shape[3]});  //whcn
+
+  float InputMin=-63.5,
+        InputMax=64,
+        WeightMin=-63.5,
+        WeightMax=64,
+        OutputMin=-127,
+        OutputMax=128;
+
+  std::pair<float, int32_t> scalesAndZp;
+
+  scalesAndZp = QuantizationParams<u_int8_t>(InputMin,InputMax);
+  std::vector<float> scalesInput={scalesAndZp.first};
+  std::vector<int32_t> zeroPointsInput={scalesAndZp.second};
+
+  scalesAndZp = QuantizationParams<u_int8_t>(WeightMin,WeightMax);
+  std::vector<float> scalesWeight={scalesAndZp.first};
+  std::vector<int32_t> zeroPointsWeight={scalesAndZp.second};
+
+  std::vector<float> scalesBias={scalesInput[0]*scalesWeight[0]};
+  std::vector<int32_t> zeroPointsBias={0};
+
+  scalesAndZp = QuantizationParams<u_int8_t>(OutputMin,OutputMax);
+  std::vector<float> scalesOutput={scalesAndZp.first};
+  std::vector<int32_t> zeroPointsOutput={scalesAndZp.second};
+
+  tim::vx::Quantization quantInput(tim::vx::QuantType::ASYMMETRIC,2,scalesInput,zeroPointsInput);
+  tim::vx::Quantization quantWeight(tim::vx::QuantType::ASYMMETRIC,2,scalesWeight,zeroPointsWeight);
+  tim::vx::Quantization quantBias(tim::vx::QuantType::ASYMMETRIC,2,scalesBias,zeroPointsBias);
+  tim::vx::Quantization quantOutput(tim::vx::QuantType::ASYMMETRIC,2,scalesOutput,zeroPointsOutput);
+
+  tim::vx::TensorSpec input_spec(tim::vx::DataType::UINT8, input_shape,
+                                 tim::vx::TensorAttribute::INPUT,quantInput);
+  tim::vx::TensorSpec weight_spec(tim::vx::DataType::UINT8, weight_shape,
+                                  tim::vx::TensorAttribute::CONSTANT,quantWeight);
+  tim::vx::TensorSpec bias_spec(tim::vx::DataType::INT32, bias_shape,
+                                tim::vx::TensorAttribute::CONSTANT,quantBias);
+  tim::vx::TensorSpec output_spec(tim::vx::DataType::UINT8, output_shape,
+                                  tim::vx::TensorAttribute::OUTPUT,quantOutput);
+
+  // Input data  nchw
+  std::vector<float> input_data_float ={
+    1, 1, 1, 1, 2, 2, 2, 2, 1, 2, 3, 4, 1, 2, 3, 4
+  };
+  // weight data   oihw
+  std::vector<float> weight_data_float = {
+    1, 2, 3, 4, -1, 1, -1, 1, -1, -1, 1, 1
+  };
+  // bias data
+  std::vector<float> bias_data_float = {
+    1, 2, 3
+  };
+  std::vector<float> golden_float = {
+    18, 18, 2, 2, 5, 5, 17, 37, 4, 4, 3, 3
+  };
+
+  std::vector<u_int8_t> input_data = Quantize<uint8_t>(input_data_float,scalesInput[0],zeroPointsInput[0]);
```

Comment:
I think it's better to use static quant data instead of dynamic calculation from the float data. Because different quantize algorithm may get different results.

Reply:
the functions in test_utils.h are general functions. 
i'll add description about the scale and ZeroPoint.